File: backend/internal/activity/dto/activity.go
package dto

import "time"

// Person 通用人员信息
type Person struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

// CreateActivityRequest 创建活动
type CreateActivityRequest struct {
	Title           string    `json:"title" binding:"required,max=200"`
	Description     string    `json:"description"`
	CoverImageID    string    `json:"cover_image_id"`
	Category        string    `json:"category" binding:"max=50"`
	Tags            []string  `json:"tags"`
	StartTime       time.Time `json:"start_time" binding:"required"`
	EndTime         time.Time `json:"end_time" binding:"required"`
	Location        string    `json:"location" binding:"max=200"`
	Latitude        *float64  `json:"latitude"`
	Longitude       *float64  `json:"longitude"`
	CheckinRadiusM  *int      `json:"checkin_radius_m"`
	MaxParticipants int       `json:"max_participants" binding:"min=0"`
}

// UpdateActivityRequest 更新活动
type UpdateActivityRequest struct {
	Title           *string    `json:"title"`
	Description     *string    `json:"description"`
	CoverImageID    *string    `json:"cover_image_id"`
	Category        *string    `json:"category"`
	Tags            []string   `json:"tags"`
	StartTime       *time.Time `json:"start_time"`
	EndTime         *time.Time `json:"end_time"`
	Location        *string    `json:"location"`
	Latitude        *float64   `json:"latitude"`
	Longitude       *float64   `json:"longitude"`
	CheckinRadiusM  *int       `json:"checkin_radius_m"`
	MaxParticipants *int       `json:"max_participants"`
	ClearGeo        bool       `json:"clear_geo"`
}

// ListActivityRequest 活动列表
type ListActivityRequest struct {
	Page      int    `form:"page"`
	PageSize  int    `form:"page_size"`
	Status    *int16 `form:"status"`
	Category  string `form:"category"`
	Keyword   string `form:"keyword"`
	StartDate string `form:"start_date"`
	EndDate   string `form:"end_date"`
}

// ActivityResponse 活动响应
type ActivityResponse struct {
	ID              string   `json:"id"`
	Title           string   `json:"title"`
	Description     string   `json:"description"`
	CoverImageID    string   `json:"cover_image_id,omitempty"`
	Category        string   `json:"category"`
	Tags            []string `json:"tags"`
	StartTime       string   `json:"start_time"`
	EndTime         string   `json:"end_time"`
	Location        string   `json:"location"`
	Latitude        *float64 `json:"latitude,omitempty"`
	Longitude       *float64 `json:"longitude,omitempty"`
	CheckinRadiusM  *int     `json:"checkin_radius_m,omitempty"`
	GPSEnabled      bool     `json:"gps_enabled"`
	MaxParticipants int      `json:"max_participants"`
	Status          int16    `json:"status"`
	Organizer       Person   `json:"organizer"`
	RegisteredCount int64    `json:"registered_count"`
	CheckedInCount  int64    `json:"checked_in_count"`
	CreatedAt       string   `json:"created_at"`
	UpdatedAt       string   `json:"updated_at"`
}

// ApproveRegistrationRequest 审批报名
type ApproveRegistrationRequest struct {
	Approve bool   `json:"approve"`
	Reason  string `json:"reason" binding:"max=500"`
}

// CheckinRequest 签到
type CheckinRequest struct {
	Method    int16    `json:"method" binding:"required,oneof=1 2"`
	Token     string   `json:"token"`
	Latitude  *float64 `json:"latitude"`
	Longitude *float64 `json:"longitude"`
}

// SurveyRequest 满意度调查
type SurveyRequest struct {
	Rating  int16  `json:"rating" binding:"required,min=1,max=5"`
	Comment string `json:"comment" binding:"max=1000"`
}

// RegistrationResponse 报名记录响应
type RegistrationResponse struct {
	ID            string `json:"id"`
	ActivityID    string `json:"activity_id"`
	User          Person `json:"user"`
	Status        int16  `json:"status"`
	CheckinStatus int16  `json:"checkin_status"`
	CheckedInAt   string `json:"checked_in_at,omitempty"`
	CheckinMethod *int16 `json:"checkin_method,omitempty"`
	CreatedAt     string `json:"created_at"`
}

// ActivityStatsResponse 活动统计
type ActivityStatsResponse struct {
	ActivityID      string           `json:"activity_id"`
	MaxParticipants int              `json:"max_participants"`
	RegisteredCount int64            `json:"registered_count"`
	ApprovedCount   int64            `json:"approved_count"`
	WaitlistCount   int64            `json:"waitlist_count"`
	CheckedInCount  int64            `json:"checked_in_count"`
	RegisterRate    float64          `json:"register_rate"`
	AttendRate      float64          `json:"attend_rate"`
	SurveyCount     int64            `json:"survey_count"`
	AvgRating       float64          `json:"avg_rating"`
	RatingDist      map[string]int64 `json:"rating_distribution"`
}

// QRCodeResponse 签到二维码
type QRCodeResponse struct {
	ActivityID  string `json:"activity_id"`
	Token       string `json:"token"`
	ExpiresAt   string `json:"expires_at"`
	CheckinPath string `json:"checkin_path"`
	PNGBase64   string `json:"png_base64"`
}

File: frontend/src/api/activity.ts
import request from './request';
import type { PageResponse } from '@/types/api';

export interface Activity {
  id: string;
  title: string;
  description: string;
  cover_image_id?: string;
  category: string;
  tags: string[];
  start_time: string;
  end_time: string;
  location: string;
  latitude?: number;
  longitude?: number;
  checkin_radius_m?: number;
  gps_enabled: boolean;
  max_participants: number;
  status: ActivityStatus;
  organizer: { id: string; name: string };
  registered_count: number;
  checked_in_count: number;
  created_at: string;
  updated_at: string;
}

export type ActivityStatus = 0 | 1 | 2 | 3 | 4;

export interface Registration {
  id: string;
  activity_id: string;
  user: { id: string; name: string };
  status: RegistrationStatus;
  checkin_status: CheckinStatus;
  checked_in_at?: string;
  checkin_method?: number;
  created_at: string;
}

export type RegistrationStatus = 0 | 1 | 2 | 3 | 4;
export type CheckinStatus = 0 | 1;

export interface ActivityStats {
  activity_id: string;
  max_participants: number;
  registered_count: number;
  approved_count: number;
  waitlist_count: number;
  checked_in_count: number;
  register_rate: number;
  attend_rate: number;
  survey_count: number;
  avg_rating: number;
  rating_distribution: Record<string, number>;
}

export interface ActivityQRCode {
  activity_id: string;
  token: string;
  expires_at: string;
  checkin_path: string;
  png_base64: string;
}

export interface CreateActivityParams {
  title: string;
  description?: string;
  cover_image_id?: string;
  category?: string;
  tags?: string[];
  start_time: string;
  end_time: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  checkin_radius_m?: number;
  max_participants?: number;
}

export interface UpdateActivityParams {
  title?: string;
  description?: string;
  cover_image_id?: string;
  category?: string;
  tags?: string[];
  start_time?: string;
  end_time?: string;
  location?: string;
  latitude?: number;
  longitude?: number;
  checkin_radius_m?: number;
  clear_geo?: boolean;
  max_participants?: number;
}

export interface ListActivityParams {
  page?: number;
  page_size?: number;
  status?: number;
  category?: string;
  keyword?: string;
  start_date?: string;
  end_date?: string;
}

export interface CheckinParams {
  method: 1 | 2;
  token?: string;
  latitude?: number;
  longitude?: number;
}

export interface SurveyParams {
  rating: number;
  comment?: string;
}

export function getActivityList(params: ListActivityParams): Promise<PageResponse<Activity>> {
  return request.get('/activities', { params });
}

export function getActivityDetail(id: string): Promise<Activity> {
  return request.get(`/activities/${id}`);
}

export function createActivity(data: CreateActivityParams): Promise<Activity> {
  return request.post('/activities', data);
}

export function updateActivity(id: string, data: UpdateActivityParams): Promise<Activity> {
  return request.put(`/activities/${id}`, data);
}

export function deleteActivity(id: string): Promise<void> {
  return request.delete(`/activities/${id}`);
}

export function startActivity(id: string): Promise<Activity> {
  return request.post(`/activities/${id}/start`);
}

export function endActivity(id: string): Promise<Activity> {
  return request.post(`/activities/${id}/end`);
}

export function cancelActivity(id: string, reason?: string): Promise<Activity> {
  return request.post(`/activities/${id}/cancel`, { reason });
}

export function registerActivity(id: string): Promise<Registration> {
  return request.post(`/activities/${id}/register`);
}

export function getMyRegistration(id: string): Promise<Registration | null> {
  return request.get(`/activities/${id}/register`);
}

export function cancelRegistration(id: string): Promise<void> {
  return request.delete(`/activities/${id}/register`);
}

export function getRegistrations(id: string): Promise<Registration[]> {
  return request.get(`/activities/${id}/registrations`);
}

export function approveRegistration(id: string, uid: string, approve: boolean, reason?: string): Promise<Registration> {
  return request.post(`/activities/${id}/registrations/${uid}/approve`, { approve, reason });
}

export function checkinActivity(id: string, data: CheckinParams): Promise<Registration> {
  return request.post(`/activities/${id}/checkin`, data);
}

export function getActivityQRCode(id: string): Promise<ActivityQRCode> {
  return request.get(`/activities/${id}/qrcode`);
}

export function getActivityStats(id: string): Promise<ActivityStats> {
  return request.get(`/activities/${id}/stats`);
}

export function submitSurvey(id: string, data: SurveyParams): Promise<void> {
  return request.post(`/activities/${id}/survey`, data);
}
